File: week02/day_2/draw_diagonal.py
__author__ = 'ylwoi'

# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was

size = int(input("Enter a number to set the diagonal size: "))

for i in range(1, size+1):
    if i == 1 or i == size:
        print(size * "%")
    else:
        print("%" + (i - 2) * " " + "%" + (size - 1 - i) * " " + "%")

File: week07/day2/draw-diagonal.js
/**
 * Created by cw on 2017-05-02.
 */
'use strict';

var lineCount = 6;
var block = '%';
var space = ' ';

// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is
for (var i = 0; i < lineCount; i++) {
    if (i === 0 || i === lineCount-1) {
        console.log(block.repeat(6));
    } else {
        console.log(block + space.repeat(i-1) + block + space.repeat(lineCount-i-2) + block);
    }
}